### DISCUSSION TITLE
iOS PAGImageView setPathAsync 无效

### DISCUSSION THREAD

**Discussion by yuximin** [Tencent/libpag#2045] [Q&A] [answered]
XCode 15.1
libpag v4.3.43
iPhoneX 16.7.2

调用PAGImageView的setPathAsync方法设置资源路径后播放无效，并且此时通过getPath()获取到的结果为空。调用PAGImageView setPath方法能正常播放。
另外，在列表中展示较小的pag特效，但是需要替换特效中的文案，此时是推荐使用PAGImageView还是PAGView，如果使用PAGImageView使用哪种方式替换图层最合适？

**Comment by kevingpqi123** [comment]
1、播放无效的问题建议直接提供复现 demo， 这样定位是最快的；
2、如果不涉及全屏播放、文本编辑和占位图替换，使用 PAGImageView，如果涉及到建议还是使用 PAGView，替换特效中的文案，渲染的结果是不会持久化存储的，这时更适合使用 PAGView， PAGImageView 适用于非编辑的场景，编辑过后主要是没有固定的缓存 key，没法缓存

**Reply by yuximin** [reply]
![image](https://github.com/Tencent/libpag/assets/37434036/460a6d5c-7216-4d0f-a10e-af863fc5b2e5)

核心代码如下：
private func startPlay() {
    let path = "https://test-static.ohlatech.com/d/5d6b4b93db3767ce671b7db94aaeb418.pag"
    self.pagImageView.setPathAsync(path) { [weak self] pagFile in
        print("资源加载成功:", pagFile.path() ?? "nil", self?.pagImageView.getPath() ?? "nil")
        self?.pagImageView.play()
    }
}

控制台输入结果：
资源加载成功: https://test-static.ohlatech.com/d/5d6b4b93db3767ce671b7db94aaeb418.pag nil

**Reply by longer-r** [reply]
官方demo就可以复现, PAGImageView的setPathAsync方法设置网络url资源无效, 另外把网络调慢, 直接堵塞UI了, 底层是用同步实现的吗

**Reply by kevingpqi123** [reply, marked as answer]
这个地方确实是 pag 的 bug，该问题今天已修复，节后我们将第一时间发布一个版本
